Modules/_json.c

```diff
@@ -1761,17 +1782,25 @@ encoder_listencode_list(PyEncoderObject *s, PyUnicodeWriter *writer,
     }
 
     if (s->markers != Py_None) {
-        int has_key;
+        Py_ssize_t len;
+
         ident = PyLong_FromVoidPtr(seq);
         if (ident == NULL)
+            return -1;
```

Comment:
```suggestion
            goto bail;
```

This will leak the converted sequence otherwise.